### PR TITLE
dynamic get all editable 'td' to update values

### PR DESCRIPTION
**Pull request by horkenw**
I was add [data-filedname] to place all json field's name for update value to use, so I can combine all field name and value to row to update table value

#2928

### CODE REVIEW

**Comment by horkenw** [comment]
and Sorry I was reinstall my laptop OS, seem I was lost all profile detail from last commit....

**Comment by djhvscf** [comment]
Do you have any fiddle where we can test it?

**Comment by horkenw** [comment]
yes , you can try out here
http://horkenw.github.io/bstclicktoedit/index.html

**Comment by djhvscf** [comment]
@horkenw let me tell you that this is a great extension but it is still presenting some issues

File: src/extensions/click-edit-row/bootstrap-table-click-edit-row.js [hunk 1]
```diff
@@ -12,11 +12,11 @@
         clickEdit: false
     });
 
-    function setDivision(node, options){
+    function setPrice(node, options){
```
Comment (djhvscf) [inline src/extensions/click-edit-row/bootstrap-table-click-edit-row.js]: This function should have a generic name

File: src/extensions/click-edit-row/bootstrap-table-click-edit-row.js [hunk 2]
```diff
@@ -12,11 +12,11 @@
...
         var $option = $('<option />');
         if(options){
             $(options).each(function(i, v){
-                $option.clone().text(v.idxNum + ' ' +v.name).val(v.idxNum).appendTo(node);
+                $option.clone().text(v.idxNum + ' ' +v.name).val(v.idxNum+v.name).appendTo(node);
```
Comment (djhvscf) [inline src/extensions/click-edit-row/bootstrap-table-click-edit-row.js]: Format this line of code `v.idxNum + v.name` (space between + sign)

File: src/extensions/click-edit-row/bootstrap-table-click-edit-row.js [hunk 3]
```diff
@@ -28,23 +28,31 @@
...
+                    fieldName.push($(td).eq(0).data('fieldname'));
+                    txt.push($(td).eq(0).find('select').val());
+                }
             });
+            // Combine field name and value for updateRow
+            updateData = txt.reduce(function(result, field, index) {
+                            result[fieldName[index]] = field;
+                            return result;
+                        }, {});
+            // console.log(updateData);
```
Comment (djhvscf) [inline src/extensions/click-edit-row/bootstrap-table-click-edit-row.js]: Delete this comment

File: src/extensions/click-edit-row/bootstrap-table-click-edit-row.js [hunk 4]
```diff
@@ -70,18 +78,21 @@
...
-                        tarNode.find('td').eq(column.fieldIndex).text('').append(select);
-                        setDivision($('#'+column.field), options);
+                        var div=$('<div class="col-md-10 col-sm-10 col-xs-10" data-fieldName='+column.field+'/>'),
+                        select=$('<select id="'+column.field+'" class="form-control" >'), options = $.selectArray[column.field];
+                        div.append(select);
+                        tarNode.find('td').eq(column.fieldIndex).text('').append(div);
+                        setPrice($('#'+column.field), options);
                         break;
                     case 'textarea':
+                        // todo
```
Comment (djhvscf) [inline src/extensions/click-edit-row/bootstrap-table-click-edit-row.js]: Delete this comment

File: src/extensions/click-edit-row/bootstrap-table-click-edit-row.js [hunk 5]
```diff
@@ -70,18 +78,21 @@
...
+                        select=$('<select id="'+column.field+'" class="form-control" >'), options = $.selectArray[column.field];
+                        div.append(select);
+                        tarNode.find('td').eq(column.fieldIndex).text('').append(div);
+                        setPrice($('#'+column.field), options);
                         break;
                     case 'textarea':
+                        // todo 
                         break;
                     default:
                         console.log(column.fieldIndex+' '+column.editable);
```
Comment (djhvscf) [inline src/extensions/click-edit-row/bootstrap-table-click-edit-row.js]: I'm not sure if we need this console.log

File: src/extensions/click-edit-row/bootstrap-table-click-edit-row.js [hunk 6]
```diff
@@ -132,10 +143,8 @@
 
         table.on('click-row.bs.table', function (e, row, $element, field) {
             if(field ==='no') return; //|| field ==='noOld'
-            this.$data.thId = $element.data().index;
-            this.$data.itemid = $element.data().uniqueid;
-            this.$data.divi = parseInt(row.area);
-            this.$data.town=parseInt(row.town);
+
+            this.$data.trId = $element.data().index;
             clikcToEdit(this, $element);
```
Comment (djhvscf) [inline src/extensions/click-edit-row/bootstrap-table-click-edit-row.js]: There is a typo `clikcToEdit` should be `clickToEdit`

**Comment by horkenw** [comment]
#2945

@djhvscf 
I'm sorry, that I just pull a new request about all that issue you ask for fix,
and is that all you mention??

**Pull request closed without merging**